internal/provider/resource_deployment.go

```diff
@@ -999,10 +1028,10 @@ func resourceDeploymentCreate(ctx context.Context, d *schema.ResourceData, meta
 
 	var diags diag.Diagnostics
 	deploymentID, err := deployer.Deploy(ctx, sub)
+	deployer.storeState(d)
```

Comment:
If the id is not set, the changes won't commit to terraform local state. So in case the contract is created, and the deployment failed, but the contract cancel failed, the resource won't be created and the contract would be leaked. It can be known whether the contract is still active or not by checking the deploymentID. If it's zero, nothing is created, otherwise there is an active contract and the id must be set.